### ISSUE TITLE
Scheduled event triggers `created_at` time shifted +04:00

### ISSUE THREAD

**Issue by grp06** [hasura/graphql-engine#5722]
I don't quite understand how nobody has reported this yet, but scheduled event triggers do not work. (Or I am somehow missing something major here)

Step 1: I create a scheduled event trigger in the Hasura UI at 17:35:04 to be fired at 17:36:25
Step 2 I create a scheduled event trigger from my server at 17:35:27 to be fired at 17:36:27

Here's a screen shot of what I see in "pending triggers"
![Screen Shot 2020-09-03 at 5 35 54 PM](https://user-images.githubusercontent.com/1573959/92176694-02f7c800-ee0d-11ea-8ce7-dbaebecd6efd.png)

The first thing to notice: look at the created_at and the "scheduled_time" ... something is off already.

Then, at 17:37:01, my webhook endpoint is hit, first with the UI-created trigger:
<img width="515" alt="Screen Shot 2020-09-03 at 5 37 49 PM" src="https://user-images.githubusercontent.com/1573959/92177287-31c26e00-ee0e-11ea-9d3e-ce01998acb78.png">

And immediately after, with the programmatically-created trigger:
<img width="651" alt="Screen Shot 2020-09-03 at 5 38 20 PM" src="https://user-images.githubusercontent.com/1573959/92177330-49015b80-ee0e-11ea-8260-5cae235c68a1.png">

Notice how the `created_at` times are incorrect.

Further confirmation of the `created_at` times being incorrect on the "processed events" tab:
![Screen Shot 2020-09-03 at 5 53 27 PM](https://user-images.githubusercontent.com/1573959/92177488-9a114f80-ee0e-11ea-976b-a4d7ac336f3e.png)

At first I thought it might have been something I'm doing wrong with Javascript dates, which are often tricky... So  here, I set the `schedule_at` time to be 4 hours ahead. In this case, I set it to be: `schedule_at 2020-09-04T01:56:34.530Z`

And here's what the pending trigger looks like in the UI:
![Screen Shot 2020-09-03 at 5 59 00 PM](https://user-images.githubusercontent.com/1573959/92177794-2d4a8500-ee0f-11ea-9c63-8ffc6caef28e.png)

I hope this is enough to help you sort this out. I replicated this on v1.3.0 and 1.3.1

**Comment by GavinRay97** [comment]
Hey, thanks for reporting this. I can replicate this, but just to confirm, the triggers do get for you at the proper time, the `created_at` is just shifted +04:00? This is the functionality I am seeing when replicating.

If so, I may edit the issue title to reflect the displayed time bug rather than broken/loss of functionality for Scheduled Triggers

**Comment by grp06** [comment]
Ok, maybe I am misunderstanding something, so I'm not sure if I can answer your question yet.  With this "1-minute sleep interval" - does this mean that I cannot guarantee that an event trigger that I specify will fire at a specific second? Like, it might not fire until the next interval "check"?

If that's the case, it seems like it's only a time display issue

**Comment by GavinRay97** [comment]
Note: it looks like this may not be a Console UI issue, the component uses the same function to render both `scheduled_time` and `created_at`. So perhaps it's getting passed in as the wrong time from the Console during creation or being stored improperly.

https://github.com/hasura/graphql-engine/blob/6cd9eb282a8afdfee41dc87bf85d09b1df85d61c/console/src/components/Services/Events/Common/Components/EventsTable.tsx#L137-L142

**Comment by grp06** [comment]
From the docs it seems like you can specify an exact time, like 6:34 pm and 35 seconds and the trigger should fire at that time. But it seems like if this interval fires every one minute and the interval is going 34 seconds past the top of the minute, this event trigger would be fired 59 seconds late?

**Comment by GavinRay97** [comment]
@grp06 

>  With this "1-minute sleep interval" - does this mean that I cannot guarantee that an event trigger that I specify will fire at a specific second? Like, it might not fire until the next interval "check"?

> this event trigger would be fired 59 seconds late?

Yeah that's precisely right, the skew can be up to 59 seconds, see:
https://github.com/hasura/graphql-engine/issues/5584#issuecomment-679940829

We should make this behavior more apparent/documented, which I believe is in the pipeline already.
I was confused by this one as well when setting up some scheduled triggers.

**Comment by grp06** [comment]
Ok, I got it. Huge bummer for me then. I was architecting my app around precisely timed event triggers. Which, if that were possible, would allow me to avoid long set timeouts and really simplify the architecture.

Thanks for your clarification.

**Comment by GavinRay97** [comment]
Absolutely, no problem. This is a not-uncommon usecase, so there's a solid chance that functionality is forthcoming at some point.

**Comment by codingkarthik** [comment]
I was not able to reproduce this issue on my local machine, but I have a fair idea of why this issue is caused. Internally, we're storing the `created_at` in a column of type `timestamp without time zone` which works strangely with the column default `now()` , which is of type `timestamp with time zone`, and when writing it into the column, it ignores the timezone part. If we change the type to `timestamp with time zone` things will work fine. (Which might explain the `+04:00` shift in `created_at`)

The second issue I found was that the `created_at` that's sent in the webhook payload, is not the `created_at` of the scheduled event, instead it's the time of creating the event invocation. 

I think the `created_at` that's sent should be of the scheduled event instead of the time of creation of the event invocation.

**Comment by tirumaraiselvan** [comment]
@grp06 This will be released in v1.4 but meanwhile you can run this SQL to mitigate the TZ shift:

```

ALTER TABLE hdb_catalog.hdb_cron_events
ALTER COLUMN created_at TYPE TIMESTAMPTZ;

ALTER TABLE hdb_catalog.hdb_cron_event_invocation_logs
ALTER COLUMN created_at TYPE TIMESTAMPTZ;

ALTER TABLE hdb_catalog.hdb_scheduled_events
ALTER COLUMN created_at TYPE TIMESTAMPTZ;

ALTER TABLE hdb_catalog.hdb_scheduled_event_invocation_logs
ALTER COLUMN created_at TYPE TIMESTAMPTZ;
```
